homeassistant/components/matter/sensor.py

```diff
@@ -549,4 +557,32 @@ def _update_from_device(self) -> None:
             clusters.ElectricalEnergyMeasurement.Attributes.CumulativeEnergyImported,
         ),
     ),
+    MatterDiscoverySchema(
+        platform=Platform.SENSOR,
+        entity_description=MatterSensorEntityDescription(
+            key="SmokeCOAlarmContaminationState",
+            translation_key="contamination_state",
+            device_class=SensorDeviceClass.ENUM,
+            state_class=None,
+            # convert to set first to remove the duplicate unknown value
+            options=list(set(CONTAMINATION_STATE_MAP.values())),
+            measurement_to_ha=CONTAMINATION_STATE_MAP.get,
+            icon="mdi:air-filter",
+        ),
+        entity_class=MatterSensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.ContaminationState,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.SENSOR,
+        entity_description=MatterSensorEntityDescription(
+            key="SmokeCOAlarmExpiryDate",
+            translation_key="expiry_date",
+            device_class=SensorDeviceClass.DATE,
+            state_class=None,
+            # raw value is epoch seconds
```

Comment:
Whats the granularity of the data? Because we also have timestamp. (not that it makes that big of a difference, but maybe if we also take the time into account, we better line up with a manufacturer for example sending a push notification that it expired)

Reply:
its seconds since epoch so this may also be a datetime from my perspective